backend/server.py

```
@@ -25,8 +26,60 @@ class QueryRequest(BaseModel):
+    """API endpoint to stream user query responses as they become available."""
+    # Handle both GET and POST requests
+    if request.method == "GET":
+        # Extract query from URL parameters
+        query_params = dict(request.query_params)
+        user_query = query_params.get("query", "")
+    else:
+        # For POST requests, extract from JSON body
+        try:
+            body = await request.json()
+            user_query = body.get("query", "")
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)